MessagingService.BusinessLogic/EventHandling/EmailDomainEventHandler.cs

```diff
@@ -0,0 +1,107 @@
+    public class EmailDomainEventHandler : IDomainEventHandler
+    {
+        #region Fields
+
+        /// <summary>
+        /// The aggregate repository
+        /// </summary>
+        private readonly IAggregateRepository<EmailAggregate> AggregateRepository;
+
+        /// <summary>
+        /// The email service proxy
+        /// </summary>
+        private readonly IEmailServiceProxy EmailServiceProxy;
+
+        #endregion
+
+        #region Constructors
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EmailDomainEventHandler"/> class.
+        /// </summary>
+        /// <param name="aggregateRepository">The aggregate repository.</param>
+        /// <param name="emailServiceProxy">The email service proxy.</param>
+        public EmailDomainEventHandler(IAggregateRepository<EmailAggregate> aggregateRepository,
+                                       IEmailServiceProxy emailServiceProxy)
+        {
+            this.AggregateRepository = aggregateRepository;
+            this.EmailServiceProxy = emailServiceProxy;
+        }
+
+        #endregion
+
+        #region Methods
+
+        /// <summary>
+        /// Handles the specified domain event.
+        /// </summary>
+        /// <param name="domainEvent">The domain event.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        public async Task Handle(DomainEvent domainEvent,
+                                 CancellationToken cancellationToken)
+        {
+            await this.HandleSpecificDomainEvent((dynamic)domainEvent, cancellationToken);
+        }
+
+        /// <summary>
+        /// Handles the specific domain event.
+        /// </summary>
+        /// <param name="domainEvent">The domain event.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        private async Task HandleSpecificDomainEvent(ResponseReceivedFromProviderEvent domainEvent,
+                                                     CancellationToken cancellationToken)
+        {
+            EmailAggregate emailAggregate = await this.AggregateRepository.GetLatestVersion(domainEvent.MessageId, cancellationToken);
+
+            // Update the aggregate with the status request information
+
+            // Get the message status from the provider
+            MessageStatusResponse messageStatus = await this.EmailServiceProxy.GetMessageStatus(domainEvent.ProviderEmailReference,
+                                                                                                domainEvent.EventCreatedDateTime,
+                                                                                                domainEvent.EventCreatedDateTime,
+                                                                                                cancellationToken);
+
+            // Update the aggregate with the response
+            switch (messageStatus.MessageStatus)
```

Comment:
Codacy found an issue: [Add a 'default' clause to this 'switch' statement.](https://app.codacy.com/manual/stuart_ferguson1/Messaging/pullRequest?prid=5953475)